lib/handler/file.c

```diff
@@ -46,6 +49,12 @@ struct st_h2o_sendfile_generator_t {
     char is_gzip;
     char send_vary;
     char *buf;
+    int range_count;
+    size_t *range_infos; /* size_t shows in pair. first is start offset, then length */
+    h2o_iovec_t boundary; /* boundary used for multipart/byteranges */
+    h2o_iovec_t mimetype; /* original mimetype for multipart */
+    int current_range; /* range that processing now */
+    int finished_range; /* range that processed till now */
```

Comment:
* please create an anonymous struct like `struct { ... } ranged;` for containing the properties necessary for handling ranged requests
* please use `size_t` for storing indices (my understanding is that `range_count`, `current_range`, `finished_range` all fit into the criteria)